File: Lesson5/Intent/app/src/main/java/com/example/intent/MainActivity.java
package com.example.intent;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void newActivity(View view) {

        Intent startIntent = new Intent(this, SecondActivity.class);

        EditText et = findViewById(R.id.etMain);
        String inputString = et.getText().toString();

        //you can also use getBaseContext() instead of this.
        startIntent.putExtra("KEY", inputString);
        startActivity(startIntent);
    }
}

File: Lesson18/Sensors/app/src/main/java/com/example/sensors/MainActivity.kt
package com.example.sensors

import android.content.Context
import android.hardware.*
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity(),SensorEventListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager

        val sensorList = sensorManager.getSensorList(Sensor.TYPE_ALL)

        Log.e("TAG", "Number of sensors ${sensorList.size}")

        sensorList.forEach {
            Log.e("TAG", it.toString())
        }

        val gravitySensor = sensorManager.getDefaultSensor(Sensor.TYPE_GRAVITY)

        sensorManager.registerListener(this, Sensor.TYPE_GRAVITY, 10000)
    }

    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onSensorChanged(event: SensorEvent?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}
